src/models/UniqueNamedContainer.java

```diff
@@ -73,13 +73,16 @@ public boolean add(T item) {
                     newItem.setUniqueName();
                 }
             }
-        }
 
-        // If the unique name is not present in the set of items then add it.
-        Optional<T> existing = findFirstByUniqueName(newItem.getUniqueName());
-        if (existing.isEmpty()) {
             items.add(newItem);
             return true;
```

Comment:
Thanks for the change here. Yet, the result of your changes triggers something else again for me to comment on (in Dutch we would call this 'mierenneuken' for which there is no direct English or Danish word).

By including a return statement inside the if-block, we could reorganize the code. Let's summarize the current code in this method as follows

```java
public boolean add(T item) {
  T newItem = (T) item.getCopy();

  if (!item.isGlobal()) {
    // Block A
  } else {
    // Block B
  }
  return false;
}
```
where block A always returns true, while block B sometimes returns true. Since block A always returns true, we could rewrite the method as
```java
public boolean add(T item) {
  T newItem = (T) item.getCopy();

  if (!item.isGlobal()) {
    // Block A
  }
  
  // If we reach here, the item is global. 
  // Block B
  return false;
}
```
Finally, since block B is much simpler than block A, I would switch the order, so you do the simple stuff first (and exit the method) or you continue with the difficult stuff.
```java
public boolean add(T item) {
  T newItem = (T) item.getCopy();

  if (item.isGlobal()) {
    // Block B
    return false;
  }
  
  // If we reach here, the item is not global. 
  // Block A
}
```